encoder/vaapiencoder_h264.cpp

```diff
@@ -927,9 +919,18 @@ Encode_Status VaapiEncoderH264::reorder(const SurfacePtr& surface, uint64_t time
 
     PicturePtr picture(new VaapiEncPictureH264(m_context, surface, timeStamp));
 
-    bool isIdr = (m_frameIndex == 0 ||m_frameIndex >= keyFramePeriod() || forceKeyFrame);
+    bool isIdr = (m_frameIndex == 0 ||m_frameIndex >= m_keyPeriod || forceKeyFrame);
 
     if (isIdr) {
+        // If the last frame before IDR is B frame, set it to P frame.
```

Comment:
maybe you should

git fetch 01org

git reset --hard 01org/master

git cherry-pick your-last-commit

git push to your github branch with force